src/main/java/matcher/Matcher.java

```diff
@@ -166,7 +169,7 @@ public void match(ClassInstance a, ClassInstance b) {
 		if (a.getArrayDimensions() != b.getArrayDimensions()) throw new IllegalArgumentException("the classes don't have the same amount of array dimensions");
 		if (a.getMatch() == b) return;
 
-		System.out.println("match class "+a+" -> "+b+(a.hasMappedName() ? " ("+a.getName(NameType.MAPPED_PLAIN)+")" : ""));
+		LOGGER.debug("Matching class {} -> {}{}", a, b, (a.hasMappedName() ? " ("+a.getName(NameType.MAPPED_PLAIN)+")" : ""));
```

Comment:
An additional note: if we change this to be debug, then expensive calls can be converted to supplier lambdas. For example, we have string concat here, so we can  convert the `(a.has...)` to a suppplier.

Reply:
Unfortunately SLF4J's API doesn't support lambdas like this directly, the only solution seems to be
```java
LOGGER.atDebug().setMessage("Matching class {} -> {}{}").addArgument(a).addArgument(b).addArgument(() -> (a.hasMappedName() ? " ("+a.getName(NameType.MAPPED_PLAIN)+")" : ""));
```
Which is quite gross, and this is a non-performance critical path anyway